src/ensembl/src/content/app/entity-viewer/gene-view/components/gene-view-sidebar/external-references/GeneExternalReferences.tsx

```diff
@@ -29,62 +32,150 @@ import ExternalReference from 'src/shared/components/external-reference/External
 import { getEntityViewerSidebarPayload } from 'src/content/app/entity-viewer/state/sidebar/entityViewerSidebarSelectors';
 
 import { RootState } from 'src/store';
-import { EntityViewerSidebarPayload } from 'src/content/app/entity-viewer/state/sidebar/entityViewerSidebarState';
-import { ExternalLink } from 'src/content/app/entity-viewer/types/externalLink';
+import {
+  CrossReference,
+  CrossReferenceGroup
+} from 'src/content/app/entity-viewer/types/crossReference';
+import { EntityViewerParams } from 'src/content/app/entity-viewer/EntityViewer';
 
 import styles from './GeneExternalReferences.scss';
 
-type Props = {
-  sidebarPayload: EntityViewerSidebarPayload | null;
+const QUERY = gql`
+  query Gene($stable_id: String!, $genome_id: String!) {
+    gene(byId: { stable_id: $stable_id, genome_id: $genome_id }) {
+      name
+      stable_id
+      cross_references {
+        id
+        name
+        description
+        url
+        source {
+          id
+          name
+          url
+        }
+      }
+      transcripts {
+        stable_id
+        cross_references {
+          id
+          name
+          description
+          url
+          source {
+            id
+            name
+            url
+          }
+        }
+      }
+    }
+  }
+`;
+
+type Gene = {
+  name: string;
+  stable_id: string;
+  transcripts: {
+    stable_id: string;
+    cross_references: CrossReference[];
+  }[];
+  cross_references: CrossReference[];
 };
 
-const GeneExternalReferences = (props: Props) => {
-  if (!props.sidebarPayload) {
+const getFormattedCrossReferences = (crossReferences: CrossReference[]) => {
```

Comment:
what does "formatted cross-references" mean?

is this function just building cross-reference groups? should it be called `buildCrossReferenceGroups` or `groupCrossReferences`?

Reply:
Renamed it to `buildCrossReferenceGroups`